src/api/draftSpecs.ts

```diff
@@ -185,18 +187,15 @@ export const getDraftSpecsBySpecType = async (
 
 export const getDraftSpecsBySpecTypeReduced = async (
     draftId: string,
-    specType: Entity,
-    fetchSpec?: boolean
+    specType: Entity
 ) => {
     const responses = await pagedFetchAll<DraftSpecsExtQuery_BySpecTypeReduced>(
         DEFAULT_PAGING_SIZE,
         'getDraftSpecsBySpecTypeReduced',
         (start) =>
             supabaseClient
                 .from(TABLES.DRAFT_SPECS_EXT)
-                .select(
-                    `draft_id,catalog_name,spec_type${fetchSpec ? ',spec' : ''}`
-                )
+                .select(`draft_id,catalog_name,spec_type,spec`)
```

Comment:
Since we need to cleanup old `reset` flags if the user switches their choice we will always need to fetch the `spec` here.